client/src/CodeClaimPage/index.tsx

```diff
@@ -46,66 +46,75 @@ export const CodeClaimPage: React.FC<RouteComponentProps<{ hash: string, method:
   }, []); /* eslint-disable-line react-hooks/exhaustive-deps */
 
   useEffect(() => {
-    if (claim && claim.delegated_mint && !isVerified) {
-      verifySignedMessage()
+    if (claim && claim.email_claimed) {
+      setIsVerified(true);
+    }
+
+    if (claim && claim.delegated_mint && !isVerified && !claim.email_claimed) {
+      verifySignedMessage();
     }
   }, [claim]); /* eslint-disable-line react-hooks/exhaustive-deps */
 
   const fetchClaim = (hash: string) => {
     setIsClaimLoading(true);
+
     getClaimHash(hash.toLowerCase())
-      .then(claim => {
+      .then((claim) => {
         setClaim(claim);
         setClaimError(false);
       })
-      .catch(error => {
+      .catch((error) => {
         setClaimError(true);
       })
       .finally(() => setIsClaimLoading(false));
-  }
+  };
 
   const continueClaim = (claim: HashClaim) => {
-    setClaim(claim)
-    setInitialStep(false)
-  }
+    setClaim(claim);
+    setInitialStep(false);
+  };
 
   const checkUserTokens = () => {
-    if (!claim || !claim.beneficiary) return
-    getTokensFor(claim.beneficiary)
-      .then(tokens => {
-        if(tokens.filter(token => token.event.id === claim.event_id).length > 0) {
-          setBeneficiaryHasToken(true)
-        }
-      })
-  }
+    if (!claim || !claim.beneficiary) return;
+    getTokensFor(claim.beneficiary).then((tokens) => {
+      if (tokens.filter((token) => token.event.id === claim.event_id).length > 0) {
+        setBeneficiaryHasToken(true);
+      }
```

Comment:
Even this was already being written we can improve it a little bit

```suggestion
      const beneficiaryHasToken = tokens.filter(({ event }) => event.id === claim.event_id).length > 0;
      if (beneficiaryHasToken) {
        setBeneficiaryHasToken(true);
      }
```